x-pack/plugins/security/common/model/authenticated_user.ts

```diff
@@ -28,9 +29,9 @@ export interface AuthenticatedUser extends User {
   lookup_realm: UserRealm;
 
   /**
-   * Name of the Kibana authentication provider that used to authenticate user.
+   * The authentication provider that used to authenticate user.
    */
-  authentication_provider: string;
+  authentication_provider: AuthenticationProvider;
```

Comment:
__note:__ this change isn't required in this PR, but I wanted make it for a long time and we'll need it in anonymous access PR anyway.